### ISSUE TITLE
isNothing is a bit inconsistent

### ISSUE THREAD

**Issue by lukigarazus** [MeetMartin/lambda#2]
`isNothing` returns `Nothing` for an empty array but it doesn't return `Nothing` for an empty object. This is a bit inconsistent for me. Why are arrays treated differently? In JS:
```
if ([]) console.log("Array is truthy")
if ({}) console.log("Object is truthy")
```

**Comment by MeetMartin** [comment]
@lukigarazus you got a point. I will change the behavior to consider empty objects as Nothing. It is an omission rather than intention on my part. I will update this issue as soon as I push the change.

Alternatively you are welcome to submit a pull request yourself and become the first official contributor :)

**Comment by lukigarazus** [comment]
With pleasure

**Comment by MeetMartin** [comment]
In my experience the fact that empty array and object are truthy confuses JavaScript developers because they expect it to be false. It would make sense for both to evaluate as Nothing in a context of Maybe result. Empty array and empty object just don't represent meaningful value.

**Comment by lukigarazus** [comment]
Sure, seems legit. Is `{}` empty then? In the code, emptiness is determined on the basis of length, and right now `{}` doesn't evaluate to empty.

**Comment by MeetMartin** [comment]
@lukigarazus {} should be empty. I had a look and other libraries consider it empty as well so we should align. I updated the code and tests accordingly and published a new version of @7urtle/lambda that has the changes in.  I also updated the relevant documentation on the website: https://www.7urtle.com/documentation-7urtle-lambda#lambda-isNothing (also affects isJust, isEmpty, and isNotEmpty). Does this solve your concern?
